### ISSUE TITLE
image label inconsistent

### ISSUE THREAD

**Issue by hseffler** [google/cadvisor#2435]
I am looking at metrics exported by cadvisor as embedded in kubelet from k8s 1.13.5.

On any random worker node the image-label seems to have one of the following values

- Just the sha256-digest
- IMAGENAME@sha256:digest
- IMAGENAME:TAG

From looking at the deployments that brought those containers there, all were deployed by tag, partly from public, mostly from a private registry.

For building queries the image-label should be consitent for all containers.

**Comment by dashpole** [comment]
This is container runtime dependent.  Which runtime are you using?  The code for docker to set this is here: https://github.com/google/cadvisor/blob/master/container/docker/handler.go#L221

**Comment by hseffler** [comment]
Yes, it's docker. Is there any reason why docker would it report the image in use so differently?

**Comment by dashpole** [comment]
Here is the docker container Config we have to work with: https://godoc.org/github.com/docker/docker/api/types/container#Config.  I assume they do that for simplicity (just return what users sent them), and also not to confuse users (e.g. "why did my image change from tag to sha when I inspect?").

**Comment by origama** [comment]
Is there any mitigation for this, is this bug going to be kept in consideration?

**Comment by dims** [comment]
Closing this issue as it has been open since before January 1, 2024. If this is still relevant, please feel free to reopen or create a new issue.
